java/console/src/main/java/Main.java

```diff
@@ -0,0 +1,267 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+
+import javax.sound.sampled.AudioFileFormat;
+import javax.sound.sampled.AudioFormat;
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+
+import com.azure.ai.openai.realtime.RealtimeAsyncClient;
+import com.azure.ai.openai.realtime.RealtimeClientBuilder;
+import com.azure.ai.openai.realtime.models.InputAudioBufferAppendEvent;
+import com.azure.ai.openai.realtime.models.RealtimeAudioInputTranscriptionModel;
+import com.azure.ai.openai.realtime.models.RealtimeAudioInputTranscriptionSettings;
+import com.azure.ai.openai.realtime.models.RealtimeRequestSession;
+import com.azure.ai.openai.realtime.models.RealtimeRequestSessionModality;
+import com.azure.ai.openai.realtime.models.RealtimeServerEventErrorError;
+import com.azure.ai.openai.realtime.models.RealtimeServerVadTurnDetection;
+import com.azure.ai.openai.realtime.models.RealtimeVoice;
+import com.azure.ai.openai.realtime.models.ResponseAudioDeltaEvent;
+import com.azure.ai.openai.realtime.models.ResponseAudioDoneEvent;
+import com.azure.ai.openai.realtime.models.ResponseAudioTranscriptDeltaEvent;
+import com.azure.ai.openai.realtime.models.ResponseAudioTranscriptDoneEvent;
+import com.azure.ai.openai.realtime.models.ServerErrorReceivedException;
+import com.azure.ai.openai.realtime.models.SessionCreatedEvent;
+import com.azure.ai.openai.realtime.models.SessionUpdateEvent;
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.credential.KeyCredential;
+import com.azure.core.util.Configuration;
+
+import reactor.core.Disposable;
+import reactor.core.Disposables;
+import reactor.core.publisher.Mono;
+
+public class Main {
+    private static final String AUDIO_RESPONSE_DATA_FILE = "audio_response.data";
+    private static final String AUDIO_RESPONSE_WAV_FILE = "audio_response.wav";
+
+    public static void main(String[] args) {
+        AudioFile audioFile = new AudioFile(Paths.get("whats_the_weather_pcm16_24khz_mono.wav"))
+                .setBytesPerSample(16)
+                .setSampleRate(24000);
+
+        RealtimeAsyncClient client = buildClient(true);
+
+        // We have a composite disposable to collect our subscriptions
+        Disposable.Composite disposables = Disposables.composite();
+
+        // Set up event consumers for our server events of interest:
+        // - SessionCreatedEvent
+        // - ResponseAudioDeltaEvent
+        // - ResponseAudioTranscriptDeltaEvent
+        disposables.addAll(Arrays.asList(
+                client.getServerEvents()
+                        .takeUntil(serverEvent -> serverEvent instanceof SessionCreatedEvent)
+                        .ofType(SessionCreatedEvent.class)
+                        .subscribe(Main::consumeSessionCreated, Main::consumeError, Main::onSessionCreatedCompleted),
```

Comment:
Since your `Main::onSessionCreatedCompleted` method is empty, you could use a different overload of the `subscribe` method that just receives the `serverEvent` handler and the error handle. I guess that could save some confusion to some. 

```suggestion
                        .subscribe(Main::consumeSessionCreated, Main::consumeError),
```

Reply:
ah, nice! I was hoping there'd be another overload like that. I'll remove the no-op.